sdk/eidos_sdk/util/stream_collector.py

```diff
@@ -64,10 +64,13 @@ async def __anext__(self):
 
 def stream_manager(stream: AsyncIterator[StreamEvent], context: StartStreamContextEvent):
     async def _iter():
-        yield context
+        yield context.model_copy()
```

Comment:
gross bug. The event is being mutated after it is yielded in the case of nested streams, so we want to copy it so we can reference it without the stream context mutating.